### PR TITLE
Make deprecation warning clearer.

### PR DESCRIPTION
**Pull request by cmarqu**

### CODE REVIEW

File: cocotb/result.py [hunk 1]
```diff
@@ -73,14 +73,14 @@ def create_error(obj, msg):
...
-        Use ``TestError(msg)`` directly instead of this function.
+        Use ``raise TestError(msg)`` directly instead of this function.
 
     Args:
         obj: Object with a log method.
         msg (str): The log message.
     """
     warnings.warn(
-        "``create_error`` is deprecated - use ``TestError(msg)`` (or any other "
+        "``create_error`` is deprecated - use ``raise TestError(msg)`` (or any other "
```
Comment (eric-wieser) [inline cocotb/result.py]: This change isn't correct

Comment (cmarqu) [reply cocotb/result.py]: Oh okay, it looked like an inconsistency.

File: cocotb/result.py [hunk 2]
```diff
@@ -73,14 +73,14 @@ def create_error(obj, msg):
     simply to avoid too many levels of nested `try/except` blocks.
 
     .. deprecated:: 1.3
-        Use ``TestError(msg)`` directly instead of this function.
+        Use ``raise TestError(msg)`` directly instead of this function.
```
Comment (eric-wieser) [inline cocotb/result.py]: This change _is_ correct.

File: cocotb/result.py (hunk 1 above)
Comment (eric-wieser) [reply cocotb/result.py]: See above- the other change is correct

**Comment by cmarqu** [comment]
Good now?

File: cocotb/result.py (hunk 2 above)
Comment (eric-wieser) [reply cocotb/result.py]: Whoops, I somehow got this backwards. Sorry! The either one is correct n

**Comment by eric-wieser** [comment]
Sorry, I think I completely misread this patch. Both the docstring and message for raise_error should include the word raise. Create_error should not change.

**Comment by cmarqu** [comment]
I think it might be correct now - I just added the word "directly", no new "raise".
If not correct, I can work on it again in the decade where Python 2 is deprecated. Cheers to that!

**Comment by eric-wieser** [comment]
Not sure this is really that much clearer, but it's no worse, so why not

**Pull request merged**